packages/contractkit/src/utils/signing-utils.ts

```diff
@@ -103,44 +105,30 @@ export function rlpEncodedTx(tx: Tx): RLPEncodedTx {
   return { transaction, rlpEncode }
 }
 
-export function signEncodedTransaction(
-  privateKey: string,
-  rlpEncoded: RLPEncodedTx
-): {
-  s: string
-  v: string
-  r: string
-} {
-  const hash = getHashFromEncoded(rlpEncoded.rlpEncode)
-  const signature = Account.makeSigner(
-    chainIdTransformationForSigning(rlpEncoded.transaction.chainId!)
-  )(hash, privateKey)
-  const [v, r, s] = Account.decodeSignature(signature)
-
-  return signatureFormatter({ v, r, s })
-}
-
 export function signatureFormatter(signature: {
   v: string
   r: string
   s: string
-}): { v: string; r: string; s: string } {
+}): { v: number; r: Buffer; s: Buffer } {
   return {
-    v: makeEven(trimLeadingZero(ensureLeading0x(signature.v))),
-    r: makeEven(trimLeadingZero(ensureLeading0x(signature.r))),
-    s: makeEven(trimLeadingZero(ensureLeading0x(signature.s))),
+    v: parseInt(makeEven(trimLeadingZero(ensureLeading0x(signature.v))), 16),
+    r: Buffer.from(makeEven(trimLeadingZero(ensureLeading0x(signature.r)))),
+    s: Buffer.from(makeEven(trimLeadingZero(ensureLeading0x(signature.s)))),
   }
 }
 
 export async function encodeTransaction(
   rlpEncoded: RLPEncodedTx,
-  signature: { v: string; r: string; s: string }
+  signature: { v: number; r: Buffer; s: Buffer }
 ): Promise<EncodedTransaction> {
   const hash = getHashFromEncoded(rlpEncoded.rlpEncode)
 
+  const v = stringNumberToHex(signature.v)
+  const r = signature.r.toString()
```

Comment:
This and s below should be something like `ensureLeading0x(signature.r.toString('hex'))`

Reply:
👍